File: js-lib/src/models/hand/Hand.ts
import { type Furo } from '../Furo.js'
import { type AbstractHandPattern, type HandPattern } from './HandPattern.js'
import { type Tile } from '../Tile.js'

export interface Hand<P extends HandPattern = AbstractHandPattern> {
  tiles: Tile[]
  furo: Furo[]
  patterns: P[]
}

File: mahjong-model/src/commonMain/kotlin/mahjongutils/models/hand/Hand.kt
package mahjongutils.models.hand

import kotlinx.serialization.Serializable
import mahjongutils.models.Melded
import mahjongutils.models.Tile

/**
 * 手牌
 */
@Serializable
data class Hand(
    override val tiles: List<Tile>,
    override val meldeds: List<Melded>
) : IHand
